radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/core/model/entities/ValidatorEntity.java

```diff
@@ -111,24 +116,46 @@ public void overwriteDataObject(
       updateValidatorFee(builder, preparedValidatorFee, config);
     } else if (dataObject instanceof ValidatorMetadata metadata) {
       updateMetadata(builder, metadata);
+      updateValidatorSystemMetadataVote(builder);
     } else if (dataObject instanceof ValidatorAllowDelegation allowDelegation) {
       updateAllowDelegation(builder, allowDelegation);
     } else if (dataObject
         instanceof com.radixdlt.api.core.openapitools.model.ValidatorSystemMetadata metadata) {
-      updateValidatorSystemMetadata(builder, metadata);
+      if (metadata.getData().isEmpty()) {
+        updateValidatorSystemMetadataVote(builder);
+      } else if (Bytes.allZeros(Bytes.fromHexString(metadata.getData()))) {
+        updateValidatorSystemMetadataWithdrawVote(builder);
+      } else {
+        throw new InvalidDataObjectException(
+            parsedDataObject,
+            "ValidatorSystemMetadata \"data\" field must either be empty (to vote) or consist of"
+                + " only zero bytes (to withdraw a vote)");
+      }
     } else {
       throw new EntityDoesNotSupportDataObjectException(this, parsedDataObject);
     }
   }
 
-  private void updateValidatorSystemMetadata(
-      TxBuilder builder,
-      com.radixdlt.api.core.openapitools.model.ValidatorSystemMetadata metadata) {
+  private void updateValidatorSystemMetadataVote(TxBuilder builder) {
     builder.down(
         com.radixdlt.application.validators.state.ValidatorSystemMetadata.class, validatorKey);
     builder.up(
         new com.radixdlt.application.validators.state.ValidatorSystemMetadata(
-            validatorKey, Bytes.fromHexString(metadata.getData())));
+            validatorKey,
+            maybeCandidateFork
+                .map(
+                    candidateFork ->
+                        CandidateForkVote.create(validatorKey, candidateFork).payload())
+                .orElseGet(HashUtils::zero256)
+                .asBytes()));
+  }
+
+  private void updateValidatorSystemMetadataWithdrawVote(TxBuilder builder) {
+    builder.down(
+        com.radixdlt.application.validators.state.ValidatorSystemMetadata.class, validatorKey);
```

Comment:
why `com.radixdlt.application.validators.state.` is necessary?

Reply:
one is an openapi dto and one the actual domain class, can't import both under and there are no aliases in java

Reply:
Is it possible to customize the name of the generated class?

Reply:
I don't know, but even if so, I'd rather keep the name as is to match the api specs

Reply:
I see. In this case, we are already importing `com.radixdlt.application.validators.state.ValidatorSystemMetadata`, we can just use `ValidatorSystemMetadata` for this class and `com.radixdlt.api.core.openapitools.model.ValidatorSystemMetadata` for the other one.

Reply:
done